### PR TITLE
fix(app): handle protocol creation failure in slideouts

### LINKED ISSUE #10569: 6.0 Feedback:  Upload a protocol that will parse on the app but will not on the robot.
### Overview

Given the attached protocol which has python 3.10 syntax.
We see that it will upload and be analyzed by the application due to the application having bundled python 3.10.
When we proceed to run nothing will happen because the POST to `/protocols` of the protocol returns 422
image

### Current Behavior

Robot selection slide out is still present and you can't proceed to run.

### Expected Behavior

Not sure what the design is but if the robot cannot parse the protocol we probably should not run the protocol.

### Steps To Reproduce

Use
[310only.py.zip](https://github.com/Opentrons/opentrons/files/8817804/310only.py.zip)
 

### Operating system

Mac

### Robot setup or anything else?

_No response_

### PR DESCRIPTION
**Pull request by vabruzzo**
# Overview
The app currently shows no error state with the protocol/robot slideout open and a failed protocol run creation from the 'proceed to run' button. This PR adds an error state to the protocol/robot card in the slideout if there is a `runCreationError`. Once another protocol/robot is selected the error is cleared.
I wanted to refactor and lift some state up from the buttons but the location of the `APIHostProvider` prevents this, fortunately there is not much prop drilling involved.

Closes #10569

# Changelog
- adds error state to `MiniCard` component
- uses `runCreationError` and `isCreatingRun` to conditionally render error states or disable the `proceed to run` button

Screen Shot 2022-06-28 at 9 38 00 AM

# Review requests
Use the protocol in the ticket and try to proceed to run from both the robot and protocol slideouts. Should see disabled button when run creation is loading and error card of state if there is an error.

# Risk assessment
Low

### CODE REVIEW

File: app/src/molecules/MiniCard/index.tsx [hunk 1]
```diff
@@ -33,12 +34,29 @@ const selectedOptionStyles = css`
...
   return (
     <Flex
       onClick={onClick}
-      css={isSelected ? selectedOptionStyles : unselectedOptionStyles}
+      css={
+        isError && isSelected
+          ? errorOptionStyles
+          : isSelected
+          ? selectedOptionStyles
+          : unselectedOptionStyles
```
Comment (koji) [inline app/src/molecules/MiniCard/index.tsx]: Maybe what Brian does would be good to keep the codebase consistency?
https://github.com/Opentrons/opentrons/pull/10895/files#diff-6f88d20c38e0a47e28c5bcedb16c229a4043c03c76f09c59bcccb63ccaadae68R52

File: app/src/molecules/MiniCard/index.tsx [hunk 2]
```diff
@@ -6,6 +6,7 @@ import { SPACING, Flex, COLORS, BORDERS } from '@opentrons/components'
 interface MiniCardProps {
   onClick: () => void
   isSelected: boolean
+  isError: boolean
```
Comment (koji) [inline app/src/molecules/MiniCard/index.tsx]: optional
If you added the info to the var name, it would be helpful to others?

File: app/src/molecules/MiniCard/index.tsx (hunk 1 above)
Comment (jerader) [inline app/src/molecules/MiniCard/index.tsx]: nested ternary is a bit hard to follow

**Review by jerader — APPROVED** [review]
LGTM! Just added a small comment. For a follow up, maybe adding a bit of test coverage to `ChooseProtocolSlideout` and `ChooseRobotSlideout`. Also just a note about design, but I was very confused at first when testing it out because you have to proceed to setup before seeing the error 😅

File: app/src/organisms/ChooseProtocolSlideout/index.tsx [hunk 3]
```diff
@@ -70,43 +74,65 @@ export function ChooseProtocolSlideout(
...
+                >
+                  {storedProtocol.mostRecentAnalysis?.metadata?.protocolName ??
+                    first(storedProtocol.srcFileNames) ??
+                    storedProtocol.protocolKey}
+                </StyledText>
+              </MiniCard>
+              {createRunError != null && isSelected && (
+                <StyledText
+                  as="label"
+                  color={COLORS.error}
```
Comment (koji) [inline app/src/organisms/ChooseProtocolSlideout/index.tsx]: This might be `errorText`?
Probably asking about this to Mel/Emily would be good.

https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=5196%3A190443

File: app/src/organisms/ChooseRobotSlideout/AvailableRobotOption.tsx [hunk 4]
```diff
@@ -20,14 +20,15 @@ interface AvailableRobotOptionProps {
...
 }
 
 export function AvailableRobotOption(
   props: AvailableRobotOptionProps
 ): JSX.Element {
-  const { robotName, robotModel, local, onClick, isSelected } = props
+  const { robotName, robotModel, local, onClick, isSelected, isError } = props
   return (
-    <MiniCard onClick={onClick} isSelected={isSelected}>
+    <MiniCard onClick={onClick} isSelected={isSelected} isError={isError}>
```
Comment (koji) [inline app/src/organisms/ChooseRobotSlideout/AvailableRobotOption.tsx]: completely optional
L53 might be `aria-label={(local ?? false) ? 'usb' : 'wifi'}`

L56 might be `name={(local ?? false) ? 'usb' : 'wifi'}`

**Review by koji — APPROVED** [review]
The changes look good to me. 🚀
I left a couple of comments but they are small things.

**Comment by vabruzzo** [comment]
> LGTM! Just added a small comment. For a follow up, maybe adding a bit of test coverage to `ChooseProtocolSlideout` and `ChooseRobotSlideout`. Also just a note about design, but I was very confused at first when testing it out because you have to proceed to setup before seeing the error 😅

Thank you. I'm going to push a few tests before merging this in this morning

File: app/src/molecules/MiniCard/index.tsx (hunk 1 above)
Comment (vabruzzo) [reply app/src/molecules/MiniCard/index.tsx]: perfect!

File: app/src/molecules/MiniCard/index.tsx (hunk 2 above)
Comment (vabruzzo) [reply app/src/molecules/MiniCard/index.tsx]: what do you mean for this? my intention was to keep this a general boolean as i imagine there will be a variety of reasons it might be in an error state

Comment (koji) [reply app/src/molecules/MiniCard/index.tsx]: Okay, that sounds good.

**Review by b-cooper — APPROVED** [review]
Changes look good 🌊
